### ISSUE TITLE
Phyton 2

### ISSUE THREAD

**Issue by shaikzakeer902-sys** [shaikzakeer902-sys/Samg-ai-agent#3]
from sklearn.ensemble import RandomForestClassifier
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score, roc_auc_score
import joblib
import os
from .utils import ensure_dir, save_json

class ModelAgent:
    def __init__(self, output_dir='results'):
        self.output_dir = output_dir
        ensure_dir(self.output_dir)

    def train_and_select(self, X_train, X_test, y_train, y_test):
        candidates = {
            'logistic': LogisticRegression(max_iter=1000),
            'rf': RandomForestClassifier(n_estimators=100)
        }

        best = None
        best_score = -1
        metrics = {}

        for name, model in candidates.items():
            model.fit(X_train, y_train)
            preds = model.predict(X_test)
            acc = accuracy_score(y_test, preds)
            try:
                proba = model.predict_proba(X_test)[:,1]
                auc = roc_auc_score(y_test, proba)
            except Exception:
                auc = None

            metrics[name] = {'accuracy': acc, 'auc': auc}

            if acc > best_score:
                best_score = acc
                best = (name, model)

        # save best model and metrics
        model_name, model_obj = best
        model_path = os.path.join(self.output_dir, f'best_model_{model_name}.pkl')
        joblib.dump(model_obj, model_path)
        save_json(metrics, os.path.join(self.output_dir, 'metrics.json'))

        return model_name, model_path, metrics
